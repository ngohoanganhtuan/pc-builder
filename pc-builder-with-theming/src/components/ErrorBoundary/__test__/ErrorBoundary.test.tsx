import { render } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

// Components
import { ErrorBoundary, ErrorBoundaryUI } from '@components';

beforeEach(() => {
  jest.spyOn(console, 'error').mockImplementation(() => jest.fn());
});

afterEach(() => {
  jest.restoreAllMocks();
});

describe('Test [ErrorBoundary] component', () => {
  test('Fallback component [ErrorBoundaryUI] should render correctly', () => {
    const component = renderer
      .create(
        <ErrorBoundaryUI
          error={{
            name: 'Error',
            message: 'Something wrong',
            stack: 'Error details...',
          }}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [ErrorBoundary] renders children component if no-error', () => {
    const component = <p>React Child</p>;
    const { getByText } = render(<ErrorBoundary>{component}</ErrorBoundary>);
    expect(getByText(/react child/i)).toBeTruthy();
  });

  test('Component [ErrorBoundary] renders fallback component [ErrorBoundaryUI] if has-error', () => {
    const CorruptedComponent = () => {
      throw new Error('This component has an error');
    };
    const { getAllByText } = render(
      <ErrorBoundary>
        <CorruptedComponent />
      </ErrorBoundary>
    );
    expect(CorruptedComponent).toThrow(/this component has an error/i);
    expect(getAllByText(/this component has an error/i)[0]).toBeVisible();
  });
});
