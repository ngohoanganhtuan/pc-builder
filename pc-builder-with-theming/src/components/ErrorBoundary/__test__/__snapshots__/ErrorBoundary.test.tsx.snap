// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [ErrorBoundary] component Fallback component [ErrorBoundaryUI] should render correctly 1`] = `
<div
  className="modal d-block"
  role="dialog"
>
  <div
    className="modal-dialog shadow"
    role="document"
  >
    <div
      className="modal-content border-0"
    >
      <div
        className="modal-header bg-danger text-white"
      >
        <h5
          className="modal-title"
        >
          Error 
          <i
            className="bi bi-emoji-frown"
          />
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <details>
          <summary>
            <span
              className="text-danger"
            >
              Error
            </span>
             : 
            Something wrong
          </summary>
          <p
            className="p-3"
          >
            <small>
              Error details...
            </small>
          </p>
        </details>
      </div>
    </div>
  </div>
</div>
`;
